### DISCUSSION TITLE
Does Tauri Mobile supports OTA update?

### DISCUSSION THREAD

**Discussion by fasidOnGit** [tauri-apps/tauri#8467] [Q&A]
I am about to start develop a Mobile app. and I came across Tauri's mobile support. Not sure how stable is to write mobile apps with Tauri at this point ? My deal-breakers would be
- OTA updates
- Location capabilities
- Camera capabilities
- _and some more vital for mobile apps_

**Comment by FabianLars** [comment]
> OTA updates

Nope. The mobile apps have the same limitations desktop apps have: They compile the frontend assets into the binary so updating them requires a binary update which requires a full app update. -> As far as i understand it, i'm fairly new to the mobile world.

> Location capabilities

Yes

> Camera capabilities

Kinda. You have access to kotlin/swift so you can do whatever you want. We also have a camera plugin work in progress [here](https://github.com/tauri-apps/plugins-workspace/pull/260) - idk what it's missing.

**Comment by AravindPrabhs** [comment]
Would it be possible to write your own update layer for the frontend ?  i.e. the app will use background tasks to get the new bundle, if any but how would I got about pointing tauri to a new bundle ? I would rather not do any renaming but keep the original bundle where it is in case of a bad update (maybe some cheap metrics/tests to trigger a revision). 

I have played around with wry and serving a static website from a file path and then using a custom protocol to communicate with the web view without the new Tauri release (this was over a year ago). I then updated the static website over git and just refreshed the WebView to get the update. So, I think it is possible but obviously Tauri abstracts over a lot !

**Comment by AravindPrabhs** [comment]
I had a dig into the source code and I think a possible strategy is writing an OtaAssets struct that takes the EmbeddedAssets in prod reverting when required. This can be assigned to the context that is passed to the builder when running the app. 

Context assets: https://github.com/tauri-apps/tauri/blob/0b79af711430934362602fb950c3e4cb5b59cf9c/crates/tauri/src/lib.rs#L381
Assets trait: https://github.com/tauri-apps/tauri/blob/0b79af711430934362602fb950c3e4cb5b59cf9c/crates/tauri/src/lib.rs#L340

**Comment by akodkod** [comment]
Hi 👋

Are there any updates? We would like to migrate from Capacitor to Tauri, but OTA is a very critical feature for us.

I would definitely help with this feature, but unfortunately, I don't know Rust. Maybe if you point me in the right direction, I can code something with the help of Claude Code.

Thank you!

**Reply by FabianLars** [reply]
Our sponsor/partner has a plugin for this (originated in their taurify offering) https://docs.crabnebula.dev/plugins/over-the-air-updates/

Note that this is bound to their Cloud offering and therefore a paid solution.

Disclaimer: CrabNebula is my employer.

**Reply by dev-itzarun** [reply]
Any Free solutions here. ?
